torch/testing/_asserts.py

```diff
@@ -253,9 +262,11 @@ def _equalize_attributes(actual: Tensor, expected: Tensor) -> Tuple[Tensor, Tens
     """Equalizes some attributes of two tensors for value comparison.
 
     If :attr:`actual` and :attr:`expected`
-    - are not on the same :attr:`~torch.Tensor.device`, they are moved CPU memory, and
+    - are not on the same :attr:`~torch.Tensor.device`, they are moved CPU memory,
```

Comment:
Nice fix

Reply:
Really this could be:

"If actual and expected are ...
- ... not the same dtype then they are promoted to a common dtype according to promote_types
- ... not the same layout then they are converted to strided tensors
- ... both sparse COO tensors but only one is coalesced then the other is coalesced"

that might save future "and" placement challenges